File: crates/mark-syntax/tests/fixtures/textmate/typescript/stress.ts
/* TypeScript stress fixture.
 * The comment spans multiple lines and contains non-ASCII: café λ🚀.
 */

type Pair<T extends string | number> = {
  left: T;
  right?: T;
};

const route = "/api/v1/café";
const routePattern = /^\/api\/(?<version>v\d+)\/[\p{Letter}-]+$/u;
const quotedAttribute = /(\w+)\s*=\s*(["']).*?\2/g;

const total = 42;
const count = 7;
const ratio = total / count / 2;
const regexThenDivision = routePattern.test(route) ? total / /\d+/.source.length : 0;

const message = `multi-line template
route=${route}
match=${routePattern.test(route) ? "✓" : "✗"}`;

export function pick<T extends string | number>(pair: Pair<T>): T {
  return pair.right ?? pair.left;
}

export const summary = `${pick({ left: "λ", right: "🚀" })}: ${message} ${ratio}`;

File: crates/mark-syntax/tests/fixtures/textmate/python/stress.py
# Python stress fixture with non-ASCII text: café λ🚀.

import re

ROUTE = re.compile(
    r"""^/api/(?P<name>[\w-]+)
    # verbose-regex comment that spans the raw triple-quoted string
    $""",
    re.VERBOSE,
)

DOC = """multi-line string
with café and "quotes"
and emoji 🚀
"""


def ratio(total: float, count: float) -> float:
    # regular comment before a division expression
    return total / count


class Greeter:
    def __init__(self, name: str = "λ") -> None:
        self.name = name

    def __str__(self) -> str:
        return f"{self.name}: {DOC.strip()} {ROUTE.match('/api/café') is not None}"
